Keyguard.d.ts

```diff
@@ -19,36 +19,38 @@ type AccountEntry = AccountInfo & {
     encryptedKeyPair: Uint8Array
 }
 
-type BasicTransactionRequest = {
-    type: TransactionType.BASIC
+type TransactionRequest = {
+    type: TransactionType.BASIC | TransactionType.EXTENDED
     sender: string
+    senderType?: Nimiq.Account.Type
+    senderLabel?: string
     recipient: string
+    recipientType?: Nimiq.Account.Type
     signer: string
+    signerLabel?: string
     value: number
     fee: number
     network: string
     validityStartHeight: number
+    extraData?: string,
+    mockKeyType?: EncryptionType
 }
```

Comment:
extraData should only exist with TransactionType.EXTENDED. Why not two dedicated types?

Reply:
I had type problems doing that. You are welcome to check out the branch and try to do it.